src/Workspaces/Core/Portable/Remote/RemoteHostSessionHelpers.cs

```diff
@@ -8,14 +8,12 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Execution;
+using Microsoft.CodeAnalysis.PooledObjects;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Remote
 {
-    /// <summary>
-    /// This will tie <see cref="Solution"/> and <see cref="RemoteHostClient.Connection"/>'s lifetime together
-    /// so that one can handle those more easily
-    /// </summary>
+    [Obsolete("Only used by Razor and LUT", error: false)]
```

Comment:
ugh.  they directly access these low level details? :(

Reply:
how are we going to track/update things to move them forward as well?

Reply:
I'm sending PR to Razor. We need to discuss the API usage with LUT as their use case is not as trivial.